Ninja compile error with MCUBoot RAM Load and Encryption

### Describe the bug

In my project using an stm32n657, specifically the `stm32n6570_dk` board, i'm trying to deploy MCUBoot with both the encryption and signing of the image for maximum security, me and collegues noticed that the build fails with a never seen before Ninja error im pasting here:

<img width="1722" height="1264" alt="Image" src="https://github.com/user-attachments/assets/9f25728b-6528-4b70-a8b8-feca706919ca" />

As you can see im trying to build the basic blinky sample with sysbuild and MCUBoot, with the following specific command:  `west build -p always -b stm32n6570_dk --sysbuild --pristine` and with this sysbuil.conf file

<img width="898" height="307" alt="Image" src="https://github.com/user-attachments/assets/a80a47bf-0da9-4ae9-9681-2215b7d9a10f" />

I understand that the error is caused by multiple cmake commands to generate the same encrypted and signed bin file but isnt this supposed to be possible to do? I dont see anything around that says that those two configurations arent supported togheter nor i understand why they wouldnt be.

### Regression

- [ ] This is a regression.

### Steps to reproduce

Build the `blinky` sample with the following `sysbuild.conf` entries:
SB_CONFIG_BOOTLOADER_MCUBOOT=y
SB_CONFIG_MCUBOOT_MODE_OVERWRITE_ONLY=y
SB_CONFIG_MCUBOOT_MODE_RAM_LOAD=y
SB_CONFIG_BOOT_ENCRYPTION=y 

using the following command:  `west build -p always -b stm32n6570_dk --sysbuild --pristine` 


### Relevant log output

```shell
CMake Error:
  Running

   '/usr/bin/ninja' '-C' '/home/fez/Desktop/Sneedphyr2/zephyr/samples/basic/blinky/build/blinky' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:704: zephyr/zephyr.slot1.signed.encrypted.bin is defined as an output multiple times

  



-- Generating done (0.3s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
CMake Error at cmake/modules/sysbuild_extensions.cmake:530 (message):
  CMake configure failed for Zephyr project: blinky

  Location: /home/fez/Desktop/Sneedphyr2/zephyr/samples/basic/blinky
Call Stack (most recent call first):
  cmake/modules/sysbuild_images.cmake:43 (ExternalZephyrProject_Cmake)
  cmake/modules/sysbuild_default.cmake:21 (include)
  /home/fez/Desktop/Sneedphyr2/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:75 (include)
  /home/fez/Desktop/Sneedphyr2/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  /home/fez/Desktop/Sneedphyr2/zephyr/share/sysbuild-package/cmake/SysbuildConfig.cmake:8 (include)
  template/CMakeLists.txt:10 (find_package)
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

Linux Ubuntu

### Additional Context

_No response_

--- a/cmake/mcuboot.cmake
+++ b/cmake/mcuboot.cmake
@@ -222,7 +222,7 @@ function(zephyr_mcuboot_tasks)
     endif()
 
     if(CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD OR CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD_WITH_REVERT)
-      list(APPEND byproducts ${output}.slot1.signed.encrypted.bin)
+      list(APPEND byproducts ${output}.slot1.signed.bin)
       set_property(GLOBAL APPEND PROPERTY extra_post_build_commands COMMAND
                    ${imgtool_sign} ${imgtool_args_alt_slot} ${output}.bin
                    ${output}.slot1.signed.bin)

